From 4b144a37d3634428691e623ef5063ef5341989e0 Mon Sep 17 00:00:00 2001
From: Tejun Heo <tj@kernel.org>
Date: Mon, 27 Nov 2023 15:13:58 -1000
Subject: [PATCH 041/135] scx: Add s/uSIZE typedefs in scx_common.h

The availability of s/uSIZE types are hit and miss. Let's always define them
in terms of stdint types. This makes life easier for the scx user repo.
---
 tools/sched_ext/scx_common.h | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/tools/sched_ext/scx_common.h b/tools/sched_ext/scx_common.h
index 0e93d6b69..7019d9f2d 100644
--- a/tools/sched_ext/scx_common.h
+++ b/tools/sched_ext/scx_common.h
@@ -7,15 +7,25 @@
 #ifndef __SCHED_EXT_COMMON_H
 #define __SCHED_EXT_COMMON_H
 
+#ifdef __KERNEL__
+#error "Should not be included by BPF programs"
+#endif
+
 #include <stdarg.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <stdint.h>
 
 #include "user_exit_info.h"
 
-#ifdef __KERNEL__
-#error "Should not be included by BPF programs"
-#endif
+typedef uint8_t u8;
+typedef uint16_t u16;
+typedef uint32_t u32;
+typedef uint64_t u64;
+typedef int8_t s8;
+typedef int16_t s16;
+typedef int32_t s32;
+typedef int64_t s64;
 
 #define SCX_BUG(__fmt, ...)							\
 	do {									\
-- 
2.43.0.232.ge79552d197

